vpr/src/route/rr_graph_uxsdcxx_serializer.h

```diff
@@ -288,7 +287,7 @@ class RrGraphSerializer final : public uxsd::RrGraphBase<RrGraphContextTypes> {
         , read_edge_metadata_(read_edge_metadata)
         , num_arch_switches_(num_arch_switches)
         , arch_switch_inf_(arch_switch_inf)
-        , segment_inf_(segment_inf)
+        , segment_inf_(rr_graph_builder->rr_segments())
```

Comment:
I believe we can remove this line as well. To confirm this is correct, can you double check:
- Where the ``segment_inf`` comes from? It is from the ``rr_segments`` in ``DeviceContext`` or the segment in architecture database ``t_arch``?

- If it is from the ``rr_segment`` in ``DeviceContext``: We can remove this line, and we should adapt all the functions in this header file to get ``rr_segments`` from ``rr_graph_builder``. Keep a short argument list is preferred.

- If it is from the segment in architecture database ``t_arch``: This line is wrong. We should not get it from ``rr_graph_builder`` but from ``t_arch``